scripts/db.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+db_folder="$(dirname $(pwd))/data"
+db_file_name="users.db"
+db_src="$db_folder/$db_file_name"
+backup_extension="backup"
+
+function validateInput() {
+	if ! [[ $1 =~ ^[A-Za-z_]+$ ]]
+		then echo "Input must be a Latin letters only"
+		exit 0
+	fi
+}
+
+function createUser() {
+	read -p "Enter user name: " user_name
+	validateInput $user_name
+
+	read -p "Enter user role: " user_role
+	validateInput $user_role
+
+	echo "$user_name, $user_role" >> "$db_src"
+}
+
+function checkDbExistance() {
+	if [ -f "$db_src" ]
+		then
+			$1
+		else
+			read -p "users.db does not exists. Do you want to create it? (y/n): " choise
+			case "$choise" in
+				y|Y)
+					> "$db_src"
+					$1
+					exit;;
+				n|N)
+					exit;;
+				*)
+					echo "Selection not valid!"
+					exit;;
+			esac
+	fi
+}
+
+function createBackup() {
+	cp "$db_src" "$db_folder/$(date +%F_%H-%M-%S)-$db_file_name.$backup_extension"
+}
+
+function restore() {
+	latest_file=$(ls -t1 "$db_folder" | sort -rn | head -n1)
+
+
+	if ! [[ "$latest_file" == *"$backup_extension"* ]]
+		then echo "No backup file found"
+		exit 0
+	fi
+
+	checkDbExistance
+
+	cp "$db_folder/$latest_file" "$db_src"
+}
+
+function find() {
+	checkDbExistance
+
+	read -p "Enter username: " user_name
+	user_list=$(grep -w "^$user_name" "$db_src")
+	if [ "$user_list" ]
+		then echo "$user_list"
```

Comment:
I think `find` command should produce the same output as `list` (e.g. numeration).